jvm-libs/json-rpc/src/main/kotlin/net/consensys/linea/jsonrpc/JsonRpcRequest.kt

```diff
@@ -20,6 +20,13 @@ interface JsonRpcRequest {
     }
 }
 
+internal data class JsonRpcRequestData(
```

Comment:
Can't we activate `KotlinModule` so we don't set those `JsonProperty` explicitly?

Reply:
Kotlin Modile shall be enabled, otherwise it will fail for simple Kotlin types I think. 
Anyway, I don't have a strong preference, I just did it this way to be aligned with `JsonRpcRequest{List/Map}Params`.

Reply:
I created a [branch](https://github.com/Consensys/linea-monorepo/pull/206/files) to test if we can remove them. Let me run an E2E tests there

Reply:
We're dealing with Vertx's Json, not Jackson directly in this case. I had to explicitly register KotlinModule on the codec in order to make it work in that PR

Reply:
Vertx uses [jackson under the hood](https://vertx.io/docs/vertx-core/java/#_mapping_between_json_objects_and_java_objects). 

Unfortunately, that change breaks tests elsewhere. I suggest to leave that improvement to another time as it's a minor thing and move forward with this.

Reply:
I fixed that as well. Will see if my fix works

Reply:
Ok, great! thanks for the contribution. 
FYI, I rebased with main.

Reply:
Tests passed

Reply:
Hi, I was keen on onboarding the changes. However, after seeing this [fix](https://github.com/Consensys/linea-monorepo/pull/206/files#diff-5b3914b01126db2aa8b6e484a6a0687af392d993970a7aedc20425fe1c3955fdR1164-R1167) I don't think it's a good idea:

- As things are (in JsonMessageProcessor, etc..), removing the annotations will require a non-obvious side-effect of setting up Vertx's Jackson codec. IMO, this is worse than having the @JsonProperty explicitly defined.
- That fix is using a deprecated API, we shall not do that - I dint't check if there is replacement. 

So, I suggest to keep the annotations for now.